raiden-ts/src/channels/utils.ts

```diff
@@ -282,6 +288,7 @@ export function approveIfNeeded$(
   tokenContract: HumanStandardToken,
   spender: Address,
   approveError: string = ErrorCodes.RDN_APPROVE_TRANSACTION_FAILED,
+  provider: JsonRpcProvider,
```

Comment:
Suggestion:
```suggestion
  { provider }: Pick<RaidenEpicDeps, 'provider'>,
```
To match `assertTx`'s signature, and make it easier to require other deps if needed in the future.